drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -284,20 +283,20 @@ static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
 		/* Send chip erase command */
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_CE);
 		return flash_npcx_nor_wait_until_ready(dev);
-	} else if (config->min_erase_size == KB(4)) {
-		opcode = SPI_NOR_CMD_SE;
-	} else if (config->min_erase_size == KB(64)) {
-		opcode = SPI_NOR_CMD_BE;
-	} else {
-		return -EINVAL;
 	}
 
 	while (size > 0) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send page/block erase command with addr */
-		flash_npcx_uma_cmd_by_addr(dev, opcode, addr);
-		addr += config->min_erase_size;
-		size -= config->min_erase_size;
+		if (size >= BLOCK_64K_SIZE) {
```

Comment:
I do not think this is completely correct: you may have offset aligned to 4k page, but not 64k page, and want to erase size of 4k + 64k, which is BLOCK_4K_SIZE followed by BLOCK_64K_SIZE.
This will be accepted by line 275, but the erase below will try to erase 64k first and will fail the entire call because the offset is not aligned to 64k.

Reply:
I am not sure I understood the comment entirely, but there is 4K check in else if right, it should capture it and do the erase if size is not aligned to 64k.

If the erase memory is continuous, the order doesn't matter right, I mean 64k block erase followed by 4k block erase is similar to 4k + 64k block erases

Reply:
> 
> If the erase memory is continuous, the order doesn't matter right (this is my assumption, I have less idea about this), I mean 64k block erase followed by 4k block erase is similar to 4k + 64k block erases, isn't it

Your assumption should at least be verified with the doc of the device the SPI controller works on the board you are using.
My assumption comes from mx25 spec that states that sector erases should have sector address and block erases should have  block address.

Reply:
oh ok Thank you

@MulinChao For this driver erase, should we make sure block erases have block aligned offset, sector erases have sector aligned offset